sdk/python/kfp/dsl/io_types.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classes for input/output types in KFP SDK.
+
+These are only compatible with v2 Pipelines.
+"""
+
+import os
+from typing import Dict, Optional, Type, TypeVar
+
+
+class Artifact(object):
```

Comment:
For my education, these classes will replace the existing ones we have, right?

Reply:
Yes, it should replace them (or merge with them). Again, I didn't change those to avoid the PR becoming too big. I will clean this up. Am discussing this with @dushyanthsc next.